packages/plugin-data/src/api/data/tree/importApi.ts

```diff
@@ -0,0 +1,496 @@
+    if (filepath.endsWith('.json')) {
+      tmpContentType = jsonContentType;
+      meta.isJson = true;
+      meta.refRegex = jsonRefRegex;
+    } else if (filepath.endsWith('.xml')) {
+      tmpContentType = xmlContentType;
+      meta.refRegex = xmlRefRegex;
+    }
+
+    // unable to determine content type from extension, was a global content type provided?
+    if (!tmpContentType) {
+      if (!contentType) {
+        const err = SfdxError.create('@salesforce/plugin-data', 'importApi', 'unknownContentType', [filepath]);
+        err.name = INVALID_DATA_IMPORT_ERR_NAME;
+        throw err;
+      } else if (contentType.toUpperCase() === 'JSON') {
+        tmpContentType = jsonContentType;
+        meta.isJson = true;
+        meta.refRegex = jsonRefRegex;
+      } else if (contentType.toUpperCase() === 'XML') {
+        tmpContentType = xmlContentType;
+        meta.refRegex = xmlRefRegex;
+      } else {
+        const err = SfdxError.create('@salesforce/plugin-data', 'importApi', 'dataFileUnsupported', [contentType]);
+        err.name = INVALID_DATA_IMPORT_ERR_NAME;
+        throw err;
+      }
+    }
+
+    meta.headers['content-type'] = tmpContentType;
+
+    return meta;
+  }
+
+  // Parse the SObject tree file, resolving any saved refs if specified.
+  // Return a promise with the contents of the SObject tree file and the type.
+  private async parseSObjectTreeFile(
+    filepath: string,
+    isJson: boolean,
+    refRegex: RegExp,
+    resolveRefs?: boolean,
+    refMap?: any
+  ): Promise<any> {
+    let contentStr: string;
+    let contentJson;
+    let match;
+    let sobject = '';
+    const foundRefs = new Set<string>();
+
+    // call identity() so the access token can be auto-updated
+    const content = await fs.readFile(filepath);
+    if (!content) {
+      throw SfdxError.create('@salesforce/plugin-data', 'importApi', 'dataFileEmpty', [filepath]);
+    }
+
+    contentStr = content.toString();
+
+    if (isJson) {
+      // is valid json?  (save round-trip to server)
+      try {
+        contentJson = JSON.parse(contentStr);
+
+        // All top level records should be of the same sObject type so just grab the first one
+        const type = getString(contentJson, 'records[0].attributes.type');
+        if (type) {
+          sobject = type.toLowerCase();
+        }
+      } catch (e) {
+        throw SfdxError.create('@salesforce/plugin-data', 'importApi', 'dataFileInvalidJson', [filepath]);
+      }
+    }
+
+    // if we're replacing references (@AcmeIncAccountId), find references in content and
+    // replace with reference found in previously saved records
+    if (resolveRefs && refMap) {
+      // find and stash all '@' references
+
+      // Could optionally use String.match(regex) to avoid turning off the rule
+      /* eslint-disable no-cond-assign */
+      while ((match = refRegex.exec(contentStr))) {
+        foundRefs.add(match[1]);
+      }
+      /* eslint-enable no-cond-assign */
```

Comment:
do we need this rule too?

Reply:
Probably not.  I'll try removing - thanks!